src/adapter/src/coord/sequencer/inner/create_index.rs

```diff
@@ -422,6 +419,10 @@ impl Coordinator {
                     coord.ship_dataflow(df_desc, cluster_id).await;
                 }
 
+                coord
```

Comment:
What's the policy where internal mutations to the in-memory catalog state should be done compared to other things that happen in the `side_effect` callback (for example before and after the dataflow is created by the compute controller)? After the change, the two places where this happens `create_materialized_view_finish` and `create_index_finish` won't agree.

Reply:
Ah, ok--my misread of this code. Will undo.